File: cash.py
from cs50 import get_float

# Prompt User for non-negative input
money = get_float("Change owed: ")
while not money > 0:
    money = get_float("Change owed: ")

# Dictionary: keys are coin values, values are the number of those coins required to provide change
change = {0.25: 0, 0.1: 0, 0.05: 0, 0.01: 0}

# For each coin in decending value order, decrement until negative, counting each time.
for coin in sorted(change, reverse=True):
    while money - coin >= 0.0:
        money = round(money - coin, 2)
        change[coin] += 1

# Print the total number of coins provided as change
print(sum(v for v in change.values()))

File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // this is the answer
    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    int cents = 0;

    // prompt user for input that is non-negative
    float dollars  = get_float("How Much Change:\n");
    while (dollars < 0)
    {
        dollars  = get_float("How Much Change:\n");
    }
    
    // convert to int to avoid rounding errors
    int change = round(dollars * 100);
    
    // decrement .25c until negative or zero
    quarters = change / 25;
    change = change % 25;
    // decrement by 0.1c until negative or zero
    dimes = change / 10;
    change = change % 10;
    // decrement by 0.05c until negative or zero
    nickels = change / 5;
    change = change % 5;
    // decrement by 0.01c until negatives or zero
    cents = change;

    printf("%i\n", quarters + dimes + nickels + cents);
}